CodeGenerator/Templates/Custom/s3/multipart_API.stencil

```diff
@@ -0,0 +1,250 @@
+                        }
+
+                        // output the data downloaded
+                        return eventLoop.submit { () -> Int64 in
+                            try outputStream(body, fileSize)
+                            return fileSize
+                        }
+
+                    } catch {
+                        return eventLoop.makeFailedFuture(error)
+                    }
+            }.flatMap { (outputBody) -> Future<Int64> in
+                let newOffset = offset + Int64(partSize)
+                return multipartDownloadPart(fileSize: fileSize, offset: newOffset)
+            }
+            return result
+        }
+
+        // get object size before downloading
+        let headRequest = S3.HeadObjectRequest(
+            bucket: input.bucket,
+            ifMatch: input.ifMatch,
+            ifModifiedSince: input.ifModifiedSince,
+            ifNoneMatch: input.ifNoneMatch,
+            ifUnmodifiedSince: input.ifUnmodifiedSince,
+            key: input.key,
+            requestPayer: input.requestPayer,
+            sSECustomerAlgorithm: input.sSECustomerAlgorithm,
+            sSECustomerKey: input.sSECustomerKey,
+            sSECustomerKeyMD5: input.sSECustomerKeyMD5,
+            versionId: input.versionId
+        )
+        let result = headObject(headRequest)
+            .hop(to: eventLoop)
+            .flatMap { object -> Future<Int64> in
+                do {
+                    guard let contentLength = object.contentLength else {
+                        throw S3ErrorType.multipart.downloadEmpty(message: "Content length is unexpectedly zero")
+                    }
+                    // download file
+                    return multipartDownloadPart(fileSize: contentLength, offset: 0)
+                } catch {
+                    return eventLoop.makeFailedFuture(error)
+                }
+        }
+        return result
+    }
+
+    /// Multipart download of a file from S3.
+    ///
+    /// - parameters:
+    ///     - input: The GetObjectRequest shape that contains the details of the object request.
+    ///     - partSize: Size of each part to be downloaded
+    ///     - filename: Filename to save download to
+    ///     - progress: Callback that returns the progress of the download. It is called after each part is downloaded with a value between 0.0 and 1.0 indicating how far the download is complete (1.0 meaning finished).
+    /// - returns: A future that will receive the complete file size once the multipart download has finished.
+    func multipartDownload(_ input: GetObjectRequest, partSize: Int = 5*1024*1024, filename: String, progress: @escaping (Double) throws -> () = { _ in }) -> Future<Int64> {
+        let eventLoop = AWSClient.eventGroup.next()
+        return eventLoop.submit {  () -> Foundation.FileHandle in
+            if FileManager.default.fileExists(atPath: filename) {
+                try FileManager.default.removeItem(atPath: filename)
```

Comment:
The FileManager.default.removeItem() is unnecesary. The subsequent createFile will clear the contents of the file.